paparazzi/src/main/java/app/cash/paparazzi/PaparazziSdk.kt

```diff
@@ -285,21 +285,29 @@ public class PaparazziSdk @JvmOverloads constructor(
       viewGroup.addView(modifiedView)
       for (frame in 0 until frameCount) {
         val nowNanos = (startNanos + (frame * 1_000_000_000.0 / fps)).toLong()
-        withTime(nowNanos) {
-          val result = renderSession.render(true)
-          if (result.status == ERROR_UNKNOWN) {
-            throw result.exception
+        var result = withTime(nowNanos) {
+          renderSession.render(true)
+        }
+
+        // If we have pendingTasks run recomposer to ensure we get the correct frame.
+        result = recomposerWatcher.onRenderResult(frame, result) {
+          withTime(nowNanos) {
+            renderSession.render(true)
           }
+        }
+
+        if (result.status == ERROR_UNKNOWN) {
```

Comment:
We can move this above the `onRenderResult` call above since that function will throw the exception on subsequent renders. This could help bail earlier if the first render is an error

Reply:
I might move this into `RecomposerWatcher.onRenderResult()` to do the same for any renders that fail.